src/coreclr/utilcode/utsem.cpp

```diff
@@ -172,6 +183,10 @@ UTSemReadWrite::Init()
 
     m_hWriteWaiterEvent = CreateEvent(NULL, FALSE, FALSE, NULL);
     IfNullRet(m_hWriteWaiterEvent);
+#else // HOST_WINDOWS
+    pthread_rwlock_init(&m_rwLock, nullptr);
```

Comment:
Do we need to handle failures?

Reply:
I think so, but unsure about what to do on failure.

Reply:
Reading through the [manpage](https://www.man7.org/linux/man-pages/man3/pthread_rwlock_rdlock.3p.html), it's unclear to me that when a writer is holding the lock, whether a reader will be blocked or returned with EBUSY. Simply returning E_FAIL under any failure?